JPS_BASE_LIB/pom.xml

```diff
@@ -10,7 +10,7 @@
 
     <!-- Please refer to the Versioning page on TheWorldAvatar wiki for
     details on how version numbers should be selected -->
-    <version>1.41.0</version>
+    <version>1.42.0-1558-sparql-federation-with-creds-SNAPSHOT</version>
```

Comment:
Using the following jps base lib version in a test agent seems to break the agent when it is deployed. The test agent was developed using the [FHSashAndOccupancyAgent](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/Agents/FHSashAndOccupancyAgent) as a template. 

I tried to use the original FHSashAndOccupancyAgent with the snapshot version of the jps base lib and when deployed, it shows the following logs in the docker container:

2023-12-20 15:45:11 NOTE: Picked up JDK_JAVA_OPTIONS:  --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.util.concurrent=ALL-UNNAMED --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED
2023-12-20 15:45:11 20-Dec-2023 07:45:11.433 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version name:   Apache Tomcat/9.0.84
2023-12-20 15:45:11 20-Dec-2023 07:45:11.436 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server built:          Dec 7 2023 19:27:40 UTC
2023-12-20 15:45:11 20-Dec-2023 07:45:11.437 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Server version number: 9.0.84.0
2023-12-20 15:45:11 20-Dec-2023 07:45:11.437 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Name:               Linux
2023-12-20 15:45:11 20-Dec-2023 07:45:11.437 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log OS Version:            5.10.16.3-microsoft-standard-WSL2
2023-12-20 15:45:11 20-Dec-2023 07:45:11.437 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Architecture:          amd64
2023-12-20 15:45:11 20-Dec-2023 07:45:11.437 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Java Home:             /opt/java/openjdk
2023-12-20 15:45:11 20-Dec-2023 07:45:11.437 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Version:           21.0.1+12-LTS
2023-12-20 15:45:11 20-Dec-2023 07:45:11.437 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log JVM Vendor:            Eclipse Adoptium
2023-12-20 15:45:11 20-Dec-2023 07:45:11.438 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_BASE:         /usr/local/tomcat
2023-12-20 15:45:11 20-Dec-2023 07:45:11.438 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log CATALINA_HOME:         /usr/local/tomcat
2023-12-20 15:45:11 20-Dec-2023 07:45:11.443 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.lang=ALL-UNNAMED
2023-12-20 15:45:11 20-Dec-2023 07:45:11.443 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.io=ALL-UNNAMED
2023-12-20 15:45:11 20-Dec-2023 07:45:11.444 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.util=ALL-UNNAMED
2023-12-20 15:45:11 20-Dec-2023 07:45:11.444 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.base/java.util.concurrent=ALL-UNNAMED
2023-12-20 15:45:11 20-Dec-2023 07:45:11.444 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED
2023-12-20 15:45:11 20-Dec-2023 07:45:11.444 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.config.file=/usr/local/tomcat/conf/logging.properties
2023-12-20 15:45:11 20-Dec-2023 07:45:11.444 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager
2023-12-20 15:45:11 20-Dec-2023 07:45:11.444 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djdk.tls.ephemeralDHKeySize=2048
2023-12-20 15:45:11 20-Dec-2023 07:45:11.445 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.protocol.handler.pkgs=org.apache.catalina.webresources
2023-12-20 15:45:11 20-Dec-2023 07:45:11.445 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dorg.apache.catalina.security.SecurityListener.UMASK=0027
2023-12-20 15:45:11 20-Dec-2023 07:45:11.445 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dignore.endorsed.dirs=
2023-12-20 15:45:11 20-Dec-2023 07:45:11.445 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcatalina.base=/usr/local/tomcat
2023-12-20 15:45:11 20-Dec-2023 07:45:11.445 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Dcatalina.home=/usr/local/tomcat
2023-12-20 15:45:11 20-Dec-2023 07:45:11.445 INFO [main] org.apache.catalina.startup.VersionLoggerListener.log Command line argument: -Djava.io.tmpdir=/usr/local/tomcat/temp
2023-12-20 15:45:11 20-Dec-2023 07:45:11.448 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent Loaded Apache Tomcat Native library [1.2.39] using APR version [1.7.0].
2023-12-20 15:45:11 20-Dec-2023 07:45:11.448 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent APR capabilities: IPv6 [true], sendfile [true], accept filters [false], random [true], UDS [true].
2023-12-20 15:45:11 20-Dec-2023 07:45:11.448 INFO [main] org.apache.catalina.core.AprLifecycleListener.lifecycleEvent APR/OpenSSL configuration: useAprConnector [false], useOpenSSL [true]
2023-12-20 15:45:11 20-Dec-2023 07:45:11.450 INFO [main] org.apache.catalina.core.AprLifecycleListener.initializeSSL OpenSSL successfully initialized [OpenSSL 3.0.2 15 Mar 2022]
2023-12-20 15:45:11 20-Dec-2023 07:45:11.717 INFO [main] org.apache.coyote.AbstractProtocol.init Initializing ProtocolHandler ["http-nio-8080"]
2023-12-20 15:45:11 20-Dec-2023 07:45:11.736 INFO [main] org.apache.catalina.startup.Catalina.load Server initialization in [450] milliseconds
2023-12-20 15:45:11 20-Dec-2023 07:45:11.765 INFO [main] org.apache.catalina.core.StandardService.startInternal Starting service [Catalina]
2023-12-20 15:45:11 20-Dec-2023 07:45:11.766 INFO [main] org.apache.catalina.core.StandardEngine.startInternal Starting Servlet engine: [Apache Tomcat/9.0.84]
2023-12-20 15:45:11 20-Dec-2023 07:45:11.775 INFO [main] org.apache.catalina.startup.HostConfig.deployWAR Deploying web application archive [/usr/local/tomcat/webapps/fh-sash-and-occupancy-agent##1.2.0.war]
2023-12-20 15:45:14 20-Dec-2023 07:45:14.327 INFO [main] org.apache.jasper.servlet.TldScanner.scanJars At least one JAR was scanned for TLDs yet contained no TLDs. Enable debug logging for this logger for a complete list of JARs that were scanned but no TLDs were found in them. Skipping unneeded JARs during scanning can improve startup time and JSP compilation time.
2023-12-20 15:45:14 20-Dec-2023 07:45:14.642 SEVERE [main] org.apache.catalina.core.StandardContext.startInternal One or more listeners failed to start. Full details will be found in the appropriate container log file
2023-12-20 15:45:14 20-Dec-2023 07:45:14.647 SEVERE [main] org.apache.catalina.core.StandardContext.startInternal Context [/fh-sash-and-occupancy-agent##1.2.0] startup failed due to previous errors
2023-12-20 15:45:14 20-Dec-2023 07:45:14.655 SEVERE [main] org.apache.catalina.loader.WebappClassLoaderBase.checkThreadLocalMapForLeaks The web application [fh-sash-and-occupancy-agent##1.2.0] created a ThreadLocal with key of type [java.lang.ThreadLocal] (value [java.lang.ThreadLocal@2e4d2aaf]) and a value of type [org.apache.logging.log4j.core.LoggerContext] (value [org.apache.logging.log4j.core.LoggerContext@42c309e9]) but failed to remove it when the web application was stopped. Threads are going to be renewed over time to try and avoid a probable memory leak.
2023-12-20 15:45:14 20-Dec-2023 07:45:14.660 INFO [main] org.apache.catalina.startup.HostConfig.deployWAR Deployment of web application archive [/usr/local/tomcat/webapps/fh-sash-and-occupancy-agent##1.2.0.war] has finished in [2,885] ms
2023-12-20 15:45:14 20-Dec-2023 07:45:14.663 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-8080"]
2023-12-20 15:45:14 20-Dec-2023 07:45:14.670 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in [2933] milliseconds
2023-12-20 15:45:14 2023-12-20 07:45:14.638 (KeyValueMap)[main] INFO - properties was not loaded 
2023-12-20 15:45:14 2023-12-20 07:45:14,650 main WARN Context destroyed before it was initialized.

Agent was initially contactable via a request like this one: "curl -X GET http://localhost:3838/fh-sash-and-occupancy-agent/status" which returns "{"description":"FHSashAndOccupancyAgent is ready."}" when using jps base lib version 1.40.0 and below.

With 1.41.0 and above, the request above returns a 404 not found.

Reply:
The hotfix done in #1079 should fix this.